hnn_core/network.py

```diff
@@ -509,8 +554,134 @@ def plot_input(self, ax=None, show=True):
             plt.show()
         return ax.get_figure()
 
-    def plot_spikes(self, ax=None, show=True):
-        """Plot the spiking activity for each cell type.
+
+class Spikes(object):
+    '''The Spikes class.
+
+    Parameters
+    ----------
+    times : list [n_trials, ] of list of float | None
```

Comment:
```suggestion
    times : list (n_trials, ) of list (n_spikes) of float, shape | None
```

size is usually a tuple. Following numpy convention. It's immutable